### ISSUE TITLE
UX Searchable and Favorite icons

### ISSUE THREAD

**Issue by Acollie** [excalidraw/excalidraw#6802]
# Idea - Selectable and searchable icons

On the icons library screen allow for
- Icons to be favorited 
- Placed into groups for example Favorite 1 or commonly used
- Allow for icons to be searched by name

Here is a mock-up I had in mind
![Screenshot 2023-07-24 at 16 07 42](https://github.com/excalidraw/excalidraw/assets/12976663/75d014a4-78e1-47d5-9edb-9e8ddf749007)

**Comment by vedmichv** [comment]
One of the most important feature for me, that really needed when you work with Architecture diagrams – and need to quick find some cloud service (for example AWS - SQS or something else)

**Comment by hjonasson** [comment]
I would like to work on the search aspect of this feature and leave `Favorites` as a separate task. My thought was as follows:

- Add a magnifying glass here https://github.com/excalidraw/excalidraw/blob/7c9cf30909c6c368407994cb25e22292b99eee5d/src/components/Sidebar/SidebarHeader.tsx#L43
- When it is clicked it opens a text box.
- That text box is used to filter the displayed items [here](https://github.com/excalidraw/excalidraw/blob/7c9cf30909c6c368407994cb25e22292b99eee5d/src/components/LibraryMenuItems.tsx)
- Communication between the two will happen via the [`SidebarPropsContext`](https://github.com/excalidraw/excalidraw/blob/7c9cf30909c6c368407994cb25e22292b99eee5d/src/components/Sidebar/common.ts#L40)

All suggestions, improvements, and pointers are appreciated @ad1992 since you were triaging this task.

**Comment by jcass8695** [comment]
@hjonasson did you get any chance to work on this?

**Comment by hjonasson** [comment]
@jcass8695 Nope. I was hoping to get feedback from maintainers before doing anything.

**Comment by ololiuhqui** [comment]
@jcass8695 honestly this is a missing feature, i would prefer to directly having the search box open because i would search for items in the library almost every time i open it.
Once you start having multiple libraries and you need something you have to search for it, otherwise you just see little unclear icons..
That's why i'm mostly making use of emojis, a quick search and i can have what i need, but adding very specific images from the library would certainly enrich what i can represent.

+1 for this feature request.

**Comment by deepu105** [comment]
This is an essential feature when having a large library. I would be happy to implment this or help @hjonasson if you are still interested

**Comment by deepu105** [comment]
@vjeux is this something that will be accepted?

**Comment by mMeijden** [comment]
this is something i'm really missing in the library. +1 for this feature request

**Comment by kaisoz** [comment]
A massive +1 for this feature. This is really missing in Excalidraw

**Comment by evgenijrenke** [comment]
+1

**Comment by kaue** [comment]
+1 really hard to differentiate some icon libraries since the preview is too small.
having a search would be the best solution.

**Comment by Cohvir** [comment]
A must have! +1

**Comment by o0ZzM4g1c** [comment]
+1000

**Comment by NahumLitvin** [comment]
how is this not the main focus of the devs. this makes exelidraw barely useable...
i have no choise as that what corp chose but hate every moment.

**Comment by Cohvir** [comment]
It's been almost a year since this feature was proposed. I hope they can implement it soon

**Comment by lhvubtqn** [comment]
+1

**Comment by MChen7209** [comment]
I hope this can be implemented soon. Aside from anything super simple, the lack of a search feature makes Excalidraw a pain to use when you have to visually scroll through hundreds of icons and find 1. Also with the massive amounts of AWS Icons, its hard to know if I am using the right one.

**Comment by rodrigosiviero** [comment]
this is the only feature that stops me from using it.  +1 for this one

**Comment by mobamoh** [comment]
+1000000

**Comment by evgenijrenke** [comment]
+1 most important feature

**Comment by nickfujita** [comment]
Rather than, or in addition to, adding a search and favorites, it would be amazing to just have the items in your Library searchable and usable via the existing Command palette. This would allow you to simply use the hotkey to pull up the Command palette (cmd + /), type the name of the component (eg. cloudfront), then hit Enter, and the component will be added to the diagram.

The advantage of this is that it can all be done easily via keyboard rather than having to move the mouse over to the library (it may even be closed), click the search area, type, move mouse down to component, then select.

<img width="1230" alt="image" src="https://github.com/excalidraw/excalidraw/assets/6342442/ed3cef7b-ff01-4c44-a6bb-5994772cfb68">

**Comment by nickfujita** [comment]
Took a stab at this an make a PR:
https://github.com/excalidraw/excalidraw/pull/7916

Test it here:
https://excalidraw-git-fork-nickfujita-command-palett-7af624-excalidraw.vercel.app/

**Comment by thixpin** [comment]
+1

**Comment by LaurentiuAndrei** [comment]
+1 any way to search for icon names would be a blessing.

**Comment by nprime496** [comment]
+1

**Comment by dmorand17** [comment]
+1000

**Comment by KristopherBorja** [comment]
+1 This feature would really make it more closer competitor to draw.io. This is one of the main complaints of user in my organisation.

**Comment by oubaydos** [comment]
+1

**Comment by arentino** [comment]
+1

**Comment by theramzay** [comment]
Any updates? This feature is essential for drawing Cloud Architecture because library contains icons from whole Azure and AWS, and you just can’t afford endless scrolling for manual icon picking. We want to migrate from `draw.io` but can’t because lack of icons search.

**Comment by bn-c** [comment]
+1

**Comment by Saymmic** [comment]
+1

**Comment by MoAl78741** [comment]
+1

**Comment by k1462015** [comment]
+1

**Comment by bilalyasin1616** [comment]
+1

**Comment by arnaucasbas** [comment]
+1

**Comment by arun-k-saini** [comment]
+1

**Comment by AlexanderArvidsson** [comment]
Please, "+1"s doesn't help and it only raises unnecessary notifications to those that are actually interested in seeing this be resolved.
Instead, you can react to the OP message with a 👍.

Better yet, if you can contribute, please do so instead of raising noise.

**Comment by kuangxiang20240501** [comment]
+1

**Comment by mmacvicar** [comment]
+1

**Comment by evgenijrenke** [comment]
+1

**Comment by jozuenoon** [comment]
+1

**Comment by iperera97** [comment]
+1

**Comment by kapeka0** [comment]
+1

**Comment by tfortimmy** [comment]
+1

**Comment by ai-Ev1lC0rP** [comment]
+1

**Comment by krykra7** [comment]
+1

**Comment by manjime83** [comment]
+1

**Comment by eviln1** [comment]
+1

**Comment by farapholch** [comment]
+1

**Comment by jvmsangkal** [comment]
+1

**Comment by carlosdelcastillo** [comment]
+1

**Comment by Christian-Klempau** [comment]
+1

**Comment by gustavolwalker** [comment]
+1

**Comment by jralfonsog-db** [comment]
+1

**Comment by mata649** [comment]
+1

**Comment by ArchishmanSengupta** [comment]
I was frustrated to not have the search functionality on excalidraw for icons

So i solved the search problem. Below is the PR link - **Do make sure to upvote 👍 + comment so that the maintainers notice this change and the PR gets merged** 

[PR](https://github.com/excalidraw/excalidraw/pull/9903)

## Search functionality in command palette by just hitting (Cmd + /)

<img width="710" height="270" alt="Image" src="https://github.com/user-attachments/assets/907b7945-cc2e-4932-a0a7-4c8b9976c1f4" />

## Search functionality in the sidebar
<img width="285" height="416" alt="Image" src="https://github.com/user-attachments/assets/d262fa55-4236-49b7-8785-78de9f869aa8" />


### if you are/were also in need of this feature, now its your turn to upvote the PR

**Comment by bigrogerio** [comment]
God bless your soul man.

**Comment by andriyor** [comment]
@Acollie I think this issue can be closed since @ArchishmanSengupta has already implemented it

**Comment by Acollie** [comment]
I agree @andriyor I think this can now be closed! 🚀
